### ISSUE TITLE
LSP Server Crashing on VS Code

### ISSUE THREAD

**Issue by cs-clarence** [tamasfe/taplo#434]
I get this error when using Even Better TOML VS Code pluggin, it's making the extension unusable.
This error appears when selecting a schema (it doesn't auto-detect schemas for some reason).

```bash
panicked at 'getrandom::getrandom() failed.: Error { internal_code: 2147483660, description: "Node.js crypto module is unavailable" }', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ahash-0.8.3/src/random_state.rs:81:51

Stack:

Error
    at PA.i.wbg.__wbg_new_693216e109162396 (/home/rencedm112/.vscode/extensions/tamasfe.even-better-toml-0.19.1/dist/server.js:2:4646)
    at wasm://wasm/00fb2f8a:wasm-function[2518]:0x2e8066
    at wasm://wasm/00fb2f8a:wasm-function[4393]:0x328a27
    at wasm://wasm/00fb2f8a:wasm-function[3073]:0x303105
    at wasm://wasm/00fb2f8a:wasm-function[1499]:0x29f415
    at wasm://wasm/00fb2f8a:wasm-function[1019]:0x263047
    at wasm://wasm/00fb2f8a:wasm-function[274]:0x155b0c
    at wasm://wasm/00fb2f8a:wasm-function[362]:0x1910fd
    at wasm://wasm/00fb2f8a:wasm-function[3286]:0x30a999
    at wasm://wasm/00fb2f8a:wasm-function[889]:0x23ea21

--- snip ---

RuntimeError: unreachable
    at wasm://wasm/00fb2f8a:wasm-function[2518]:0x2e8128
    at wasm://wasm/00fb2f8a:wasm-function[4393]:0x328a27
    at wasm://wasm/00fb2f8a:wasm-function[3073]:0x303105
    at wasm://wasm/00fb2f8a:wasm-function[1499]:0x29f415
    at wasm://wasm/00fb2f8a:wasm-function[1019]:0x263047
    at wasm://wasm/00fb2f8a:wasm-function[274]:0x155b0c
    at wasm://wasm/00fb2f8a:wasm-function[362]:0x1910fd
    at wasm://wasm/00fb2f8a:wasm-function[3286]:0x30a999
    at wasm://wasm/00fb2f8a:wasm-function[889]:0x23ea21
    at wasm://wasm/00fb2f8a:wasm-function[1409]:0x295ce4
[Info  - 3:53:39 PM] Connection to server got closed. Server will restart.
```

## VS Code build info
Version: 1.80.0
Commit: 660393deaaa6d1996740ff4880f1bad43768c814
Date: 2023-07-04T13:39:48.236Z
Electron: 22.3.14
ElectronBuildId: 21893604
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Linux x64 6.3.11-200.fc38.x86_64

**Comment by zmtq05** [comment]
I switched to version 0.19.0 to avoid this issue

**Comment by tamasfe** [comment]
It's not the first time I see this, unfortunately I don't remember what caused it.

Since the vscode marketplace doesn't allow yanking versions, I re-released the previous version (`0.19.0`) as `0.19.2` for the time being.

**Comment by ngdangtu-vn** [comment]
> It's not the first time I see this, unfortunately I don't remember what caused it.

In my case it is because of schema. If I ever use it, the LSP got crash. At first it didn't, but after I tried (I'm curious ok?) draft-07 from here https://json.schemastore.org/hugo.json. The plugin went crazy. I also tried to uninstall and reinstall but there's nothing change...

Do you happen to cache your plugin somewhere in vscode?

**Comment by cs-clarence** [comment]
> It's not the first time I see this, unfortunately I don't remember what caused it.
> 
> ~Since the vscode marketplace doesn't allow yanking versions, I re-released the previous version (`0.19.0`) as `0.19.2` for the time being.~ Welp, the CD apparently ignores the tag version, so just tagging the old commit wasn't enough.

If it helps, the neovim taplo lsp works perfect.

**Comment by panekj** [comment]
> It's not the first time I see this, unfortunately I don't remember what caused it.

@tamasfe I suppose this?

https://docs.rs/getrandom/latest/getrandom/#webassembly-support
https://github.com/tamasfe/taplo/commit/61d54a8f9d8015894b7aee0435bd5dba85e49a23

**Comment by tamasfe** [comment]
Thanks, we do have the `js` feature enabled, but this could be it: https://docs.rs/getrandom/latest/getrandom/#nodejs-es-module-support.

**Comment by flying-sheep** [comment]
Fixed in #462. In order to use the workaround you linked, we need to have the version of getrandom pulled in that has the code enabling that workaround. For that I bumped our `ahash` dependencies, which in turn allowed bumping the `getrandom` version in the lockfiles.
